src/components/FeatureThresholdPanel.tsx

```diff
@@ -47,30 +83,70 @@ const defaultProps: Partial<FeatureThresholdPanelProps> = {
  * A React component for adding, removing, and editing thresholds on features in a dataset.
  */
 export default function FeatureThresholdPanel(inputProps: FeatureThresholdPanelProps): ReactElement {
-  const props = { ...defaultProps, ...inputProps } as Required<FeatureThresholdPanelProps>;
+  const props = {
+    ...defaultProps,
+    ...inputProps,
+  } as Required<FeatureThresholdPanelProps>;
 
-  // Clear thresholds for features that don't exist when the dataset changes.
-  // TODO: Show these thresholds as disabled, rather than removing them.
-  useEffect(() => {
-    const newThresholds = props.featureThresholds.filter((t) => props.dataset?.featureNames.includes(t.featureName));
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const selectContainerRef = useRef<HTMLDivElement>(null);
+
+  /** Converts a threshold to a unique key that can be used to look up its information later. Matches on feature name and unit. */
+  const thresholdToKey = (threshold: FeatureThreshold): string => {
+    return `${encodeURIComponent(threshold.featureName)}:${threshold.unit ? encodeURIComponent(threshold.unit) : ""}`;
+  };
+  // Save the min/max values of each threshold in case the user switches to a dataset that no longer has the threshold's
+  // feature (no match for name + unit). This allows the sliders to be rendered with the same bounds as before until the user switches
+  // back to a dataset that has the thresholded feature.
+  const featureMinMax = useRef<Map<string, [number, number]>>(new Map());
```

Comment:
does this contain the whole list of what is visible in the panel, including the greyed-out ones?  (it seems like this info is already saved in featureThresholds)

Reply:
Yes! This holds the absolute min/max range of the feature (essentially, the range shown on the slider) and not the min/max of the threshold. This prevents the slider bounds from jumping around as different datasets are loaded.

I'll update the comment, but let me know if it's still unclear.

Reply:
makes sense, thanks!